assets/js/atomic/blocks/product-elements/add-to-cart/product-types/variable/variation-attributes/attribute-picker.js

```diff
@@ -19,86 +20,72 @@ import {
  */
 const AttributePicker = ( { attributes, variationAttributes } ) => {
 	const [ variationId, setVariationId ] = useState( 0 );
-
-	// @todo Support default selected attributes in Variation Picker.
-	const [ selectedAttributes, setSelectedAttributes ] = useState( [] );
+	const [ selectedAttributes, setSelectedAttributes ] = useState( {} );
+	const { dispatchActions } = useAddToCartFormContext();
 
 	const attributeNames = Object.keys( attributes );
-	const hasSelectedAttributes =
-		Object.values( selectedAttributes ).filter( Boolean ).length > 0;
 	const hasSelectedAllAttributes =
 		Object.values( selectedAttributes ).filter(
 			( selected ) => selected !== ''
 		).length === attributeNames.length;
 
-	// Gets valid attribute options for the picker taking current selections into account.
+	// Get options for each attribute picker.
 	const filteredAttributeOptions = useMemo( () => {
-		const options = [];
+		return getActiveSelectControlOptions(
+			attributes,
+			variationAttributes,
+			selectedAttributes
+		);
+	}, [ attributes, variationAttributes, selectedAttributes ] );
```

Comment:
The calculation logic does loop a lot and does seem computational heavy. Perhaps it would be better to use a ref to track attributes because they don't change often.